docs: improve the doc about Raw operator and add missing test

tychota opened the pull request:
Closes https://github.com/typeorm/typeorm/issues/6860
(see the discussion there)

# What

- Improves the doc after the recent merge of https://github.com/typeorm/typeorm/pull/6850 

> (feat: add ability for escaping for Raw() find operator)

- add a missing test

# Test plan

- no code change
- waiting for CI to let new test pass
- I didn't test my changes on website as I don't know how to run the site locally: let me know if that is needed

# Note 

- I'm not a native English speaker so feel free to comment to improve the wordings, if needed, and I will update the PR.
- maybe we shall merge this once (or just before) the code is deployed on npm.

tychota commented:
@temnov98 : you may want to review this.

tychota commented:
@pleerock Out of curiosity, where does the `:...data` syntax come from ? Is it typeorm magie, or rather pg-node standard.
I found https://github.com/brianc/node-postgres/issues/1490 but I didn'tthink the discussion ended in this solution.

temnov98 commented on test/functional/repository/find-options-operators/repository-find-operators.ts:
```diff
@@ -605,6 +605,16 @@ describe("repository > find options > operators", () => {
+
+        // check operator
+        const result6 = await connection.getRepository(Post).find({
+            likes: Raw((columnAlias) => `${columnAlias} IN (:...values)`, { value: [2, 3, 6] }),
```
You have typo in parameters. Should be { values: [2, 3, 6] }

tychota replied:
I did fix it and edit the previous commit (git fixup).

Now the tests are green ✅

temnov98 commented:
> * I didn't test my changes on website as I don't know how to run the site locally: let me know if that is needed

It would be nice. You can test on at least some databases. It's not that hard - you can run databases in docker.

pleerock commented:
>  Out of curiosity, where does the :...data syntax come from ?

It is typeorm magic. Not a great implementation however, but it just works.

tychota commented:
> It would be nice. You can test on at least some databases. It's not that hard - you can run databases in docker.

@temnov98: Not sure my sentence was clear: my changes are double:
- part 1) I added documentation
- part 2) I added one test.

So:
- ✅ For 2), the CI seems to run all the tests on different env.
- ⚠️ For 1), I tried my best to write markdown, but I didn't actually test that the website is well-formated after the local changes, as I don't know how to run the website locally.

If I need to test the website locally, I expect some contributor steps like:
- clone `https://github.com/typeorm/typeorm.github.io`
- inside a subdir, clone ``https://github.com/typeorm/typeorm`
- run `yarn start`

temnov98 commented:
> * For 1), I tried my best to write markdown, but I didn't actually test that the website is well-formated after the local changes, as I don't know how to run the website locally.

Oh. I'm not a native English speaker either. Try click on this button for display the rich diff. Did you mean it?

![image](https://user-images.githubusercontent.com/22731353/95252729-0aecc280-0826-11eb-9271-f5982baf77e6.png)
![image](https://user-images.githubusercontent.com/22731353/95253239-c877b580-0826-11eb-8158-b3bde8a4f0b0.png)

tychota commented:
> Try click on this button for display the rich diff. Did you mean it?

Makes sense.
So I guess it is ready for review.

imnotjames commented on docs/find-options.md:
````diff
@@ -367,8 +367,39 @@ will execute following query:
+SELECT * FROM "post" WHERE "currentDate" > '2020-10-06'
+```
+
+If you need to provide an user input that is an array, you can unwrap the native using the following syntax:
````
We don't need the particle `an` in `an user input` - it's just `user input`.

We could probably borrow from the [`query-builder`](https://github.com/typeorm/typeorm/blob/master/docs/select-query-builder.md#using-parameters-to-escape-data) docs some of the wording, as well.

```suggestion
If you need to provide user input that is an array, you can bind them as a list of values in the SQL statement by using the special expression syntax:
```

tychota replied:
Fix and commit (doc:) amended

imnotjames commented on docs/find-options.md:
```diff
@@ -367,8 +367,39 @@ will execute following query:
 
-> Note: beware with `Raw` operator. It executes pure SQL from supplied expression and should not contain a user input,
- otherwise it will lead to SQL-injection.
+If you want to provide user input, in order to prevent SQL-injection, you can use the native parameters by providing a second argument to the `Raw` function.
```
A little more verbose but maybe clearer?  What do you think?

```suggestion
If you need to provide user input, you should not include the user input directly in your query as this may create a SQL injection vulnerability.  Instead, you can use the second argument of the `Raw` function to provide a list of parameters to bind to the query.
```

tychota replied:
Fixed and commit (doc:) amended

imnotjames commented on docs/find-options.md:
````diff
@@ -367,8 +367,39 @@ will execute following query:
+});
+```
+
+it will execute following query:
````
Just to match the previous examples above

```suggestion
will execute following query:
```

tychota replied:
Fixed and commit (doc:) amended

imnotjames commented on test/functional/repository/find-options-operators/repository-find-operators.ts:
```diff
@@ -605,6 +605,16 @@ describe("repository > find options > operators", () => {
+
+        // check operator
+        const result6 = await connection.getRepository(Post).find({
+            likes: Raw((columnAlias) => `${columnAlias} IN (:...values)`, { values: [2, 3, 6] }),
```
:raised_hands: Heck yeah, code coverage! :)

tychota replied:
👍

imnotjames reviewed (approved):
![](https://media.tenor.com/images/0e47f9eb456b11e32dccd99047591a51/tenor.gif)

tychota commented:
Thank you @imnotjames.

I think we should also deploy an npm version else the doc will document an unreleased version, as I mentioned in the PR description.
